sotodlib/core/resources.py

```diff
@@ -83,7 +86,8 @@ def get_local_file(filename: str, cache: bool = True) -> str:
         target_path = local_cache if cache else "/tmp/"
         os.makedirs(name=target_path, exist_ok=True)
         target_file = os.path.join(target_path, filename)
-        _, headers = urlretrieve(de_url, target_file)
+        if download:
+            _, headers = urlretrieve(de_url, target_file)
```

Comment:
If download is False, this function will return a filepath, but the file will not exist. Is it just for testing purposes?

Reply:
Instead of that in the test you can do:
```
from unittest import mock

@mock.patch('urllib.request.urlretrieve', return_value=["some_answer", "some_header"])
def test_get_local_file(self, mocked_urlretrieve):
...
```

and pytest will never download anything.

Reply:
Ahh, that is powerful.  However I am going to keep the download arg because it's generally useful; you might want to basically assert that a bunch of files are available.  Also if someone changed the implementation (to not use urlretrieve), this mock would become ineffective, with no warning.